containers/daap-python-base/src/var/task/data_product_metadata.py

```diff
@@ -332,23 +418,71 @@ class to handle creation and updating of
     schema json files relating to data products as a whole
     """
 
-    # returns path of latest metadata or v1 if it doesn't exist
+    MINOR_UPDATE_FIELDS = {
+        "description",
+        "email",
+        "dataProductOwner",
+        "dataProductOwnerDisplayName",
+    }
+
     def __init__(
         self,
         data_product_name: str,
         logger: DataPlatformLogger,
         input_data: dict | None,
+        version: str | None = None,
     ):
-        bucket_path = DataProductConfig(data_product_name).metadata_path()
+        bucket_path = DataProductConfig(data_product_name).metadata_path(
+            version=version
+        )
 
         super().__init__(
+            json_type=JsonSchemaName("metadata"),
             data_product_name=data_product_name,
             logger=logger,
-            json_type=JsonSchemaName("metadata"),
             bucket_path=bucket_path,
             input_data=input_data,
         )
 
+    def create_new_version(self) -> DataProductMetadata | None:
+        """
+        Create a new version of the data product (minor update only for now)
+        """
+        if not self.valid:
+            return None
+
+        self.load()
+        state = self._detect_update_type()
+
+        if state == UpdateType.MinorUpdate:
+            new_version = self.generate_next_version_string()
+            self.logger.info(f"Creating {self.data_product_name}:{new_version}")
+            new_metadata = DataProductMetadata(
+                data_product_name=self.data_product_name,
+                logger=self.logger,
+                input_data=self.data,
+                version=new_version,
+            )
+            new_metadata.write_json_to_s3()
+            return new_metadata
```

Comment:
might be personal preference but i kind of liked enforcing the write method to be passed a path, and think i'd find something like below clearer
```suggestion
            new_version_key = (
                DataProductConfig(self.data_product_name).metadata_path(new_version).key
            )
            self.write_json_to_s3(new_version_key)
            return self.data

Reply:
Fair enough, I'll just drop that commit then, as this is what it looked like before.

It makes me wonder though if we should lose the `bucket_path` parameter altogether, so that `load` is consistent with `write_json_to_s3` (both are called with a path)?

Reply:
as in the get rid of `latest_version_key`? 

I think `load` could maybe be more suitably named, like `load_latest_version` but if there might be any use for loading in older versions then possibly but i can't think of any at the moment, otherwise we'd always want to load the `latest_version_key`